src/sdam/server.ts

```diff
@@ -348,8 +349,11 @@ export class Server extends TypedEventEmitter<ServerEvents> {
       (err, conn, cb) => {
         if (err || !conn) {
           this.s.operationCount -= 1;
+          if (!err) {
+            return cb(new MongoRuntimeError('Failed to create connection without error'));
+          }
           if (!(err instanceof PoolClearedError)) {
-            markServerUnknown(this, err);
+            this.handleError(err);
           } else {
             err.addErrorLabel(MongoErrorLabel.RetryableWriteError);
           }
```

Comment:
minor: flip the if stmt, we've before preferred to keep instanceof inside an if truthy cus it's easier to read, but just an 🧅

Reply:
this isn't new logic, we already talked about the if statement on the PR that introduced it; iirc your opinion was ultimately that we should be comfortable reading inverted instanceof checks ;)